packages/playwright/src/takeSnapshot.ts

```diff
@@ -35,7 +35,15 @@ async function takeSnapshot(
   // Serialize and capture the DOM
   const domSnapshot: elementNode = await page.evaluate(dedent`
     ${rrweb};
-    rrwebSnapshot.snapshot(document);
+    if (typeof define === "function" && define.amd) {
+      new Promise((resolve) => {
+        require(['rrwebSnapshot'], (rrwebSnapshot) => {
```

Comment:
Basically, it is simply loading `rrwebSnapshot` asynchronously. This seems to be required if the page supports and uses AMD, though I don't exactly understand why. I reached out in Slack in case anyone else had more experience with this.

Reply:
What does `rrwebSnapshot` actually mean here? Is it a shorthand for `./rrwebSnapshot.js`? Is it something from the internet (npm, or?) I don't really understand what's happening under the hood here.

Reply:
On line 7, you'll see that we read in the entire contents of the rrweb-snapshot export:

```ts
const rrweb = readFileSync(require.resolve('@chromaui/rrweb-snapshot'), 'utf8');
```

This file defines `rrwebSnapshot`. Here are the first 13 lines:

```js
(function (g, f) {
    if ("object" == typeof exports && "object" == typeof module) {
      module.exports = f();
    } else if ("function" == typeof define && define.amd) {
      define("rrwebSnapshot", [], f);
    } else if ("object" == typeof exports) {
      exports["rrwebSnapshot"] = f();
    } else {
      g["rrwebSnapshot"] = f();
    }
  }(this, () => {
var exports = {};
var module = { exports };
```

Reply:
`@chromaui/rrweb-snapshot` is our fork of `rrweb-io/rrweb-snapshot`